doc/Makefile

```diff
@@ -15,41 +15,33 @@ PAPEROPT_letter = -D latex_paper_size=letter
 ALLSPHINXOPTS   = -d $(BUILDDIR)/doctrees $(PAPEROPT_$(PAPER)) $(SPHINXOPTS) .
 
 .PHONY: help clean html dirhtml pickle json htmlhelp qthelp latex changes linkcheck doctest latexpdf htmlzip
-.PHONY: all install pdf
+.PHONY: all install pdf gallery
 
-html:
+html:  gallery
 	cp sphinx/ext_mathjax.py extensions.py
-	./doc_examples_to_gallery.py
 	$(SPHINXBUILD) -b html $(ALLSPHINXOPTS) $(BUILDDIR)/html
-	cp -r examples $(BUILDDIR)/html
 	@echo
 	@echo "html build finished: $(BUILDDIR)/html."
-	rm -f *.dat *.sav *.csv
```

Comment:
not removing these files leaves untracked *.dat and *.csv files in ```doc/```, which I don't really like. We can of course add them to ```.gitignore```, which should work (similarly to what I already did for ```examples/documentation```. 

I personally don't really like to keep this stuff around, but that's the only way one can avoid building the documentation/generating the gallery everytime from scratch. So I am fine with doing this, as long as we make sure that none of these get actually incorporated into the GitHub and PyPI releases - that likely requires some changes to ```MANIFEST.in``` as well (I think the ones below, but please double-check):

change: ```recursive-include examples *.py *.ipynb *.dat```
to ```recursive-include examples *.py *.dat *.csv```

add: ```recursive-exclude examples/documentation```

change: ```recursive-exclude doc *.pdf```
to : ```recursive-exclude doc *.pdf *.dat *.csv```

Reply:
Ah, good point. 

I don't actually have a strong preference for generating gallery "only when needed" vs "every time the docs are rendered", I was just impatient ;).   

I thought it was a little odd that the `.dat`+ files needed to be in the top-level doc/ folder instead of doc/examples.    But, IIUC (please correct!) that's because that is the current working directory when the gallery is generated.  

We could consider changing all the examples that read files from disk to instead either read them from a web resource (say github), or from a (currently non-existent) `lmfit.data` folder,  or from a path relative to the example script itself.     I don't have a strong preference, but maybe that would make this all a bit easier in the long run.

Anyway, I will update `.gitignore` and `MANIFEST.in`.

Reply:
let's go for this option now as it works as intended, but we can certainly improve a bit on this. 

Currently, we need the data files both in the top-level ```doc``` folder and the ```doc/examples``` directory. The latter one is needed for the "Examples from the documentation" in the gallery, whereas the first one is needed for generating the ```example_<filname>``` examples. Of course one could use relative paths when referring to these files, but I kind of didn't like the aesthetic of it ;) But I agree, it might be worthwhile putting these data files in a separate ```lmfit.data``` directory.